### PR TITLE
Adding a Migration guide and integration article

### PR DESCRIPTION
**Pull request by rkoron007**

### CODE REVIEW

File: migration.mdx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+### Node.js
+Apollo Server 4 supports Node.js 14 and later. This includes all LTS and Current versions at the time of release.
+
+If you're using an older version of Node.js, upgrade your runtime before upgrading to Apollo Server 4.
+
+### `graphql`
+
+Apollo Server has a peer dependency on `graphql` (the core JS GraphQL implementation), which means you are responsible for choosing the version installed in your app.
+
+Apollo Server 4 supports graphql v16.3.0 and later. (Apollo Server 3 supported `graphql` v15.3.0 through v15.8.0.)
```
Comment (glasser) [inline migration.mdx:25]: AS3 supports v16 too.

**Review by glasser — APPROVED** [review]
this seems good enough to put on the branch!

File: packages/server/src/plugin/landingPage/default/types.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
+export interface ApolloServerPluginLandingPageDefaultBaseOptions {
```
Comment (glasser) [inline packages/server/src/plugin/landingPage/default/types.ts:1]: Hmm, not sure what happened here — will revert.

Comment (glasser) [reply packages/server/src/plugin/landingPage/default/types.ts:1]: Seems to be an oddity of how GH is displaying this — it's not in the squash commit.

Comment (rkoron007) [reply packages/server/src/plugin/landingPage/default/types.ts:1]: That's bizarre—nice catch though!

**Pull request merged**